hasher-matcher-actioner/src/OpenMediaMatch/blueprints/matching.py

```diff
@@ -151,11 +151,13 @@ def lookup_signal(signal: str, signal_type_name: str) -> list[int]:
 def lookup_signal_with_distance(
     signal: str, signal_type_name: str
 ) -> list[MatchWithDistance]:
+    banks = lookup(signal, signal_type_name)
     results = query_index(signal, signal_type_name)
```

Comment:
blocking: This is essentially causing us to double query to the index, since lookup will query the index, and query_index also queries the index.

Instead, what you want to be doing is taking the returned content_ids and doing a lookup on which (singular) bank they are in, which you can essentially see how to do in the body of lookup.